### ISSUE TITLE
Exclude not working as expected

### ISSUE THREAD

**Issue by mockdeep** [rubocop/rubocop#1022]
I'm trying to exclude a bunch of files from the HashSyntax cop and running into some issues. My config looks like this:

``` yaml
HashSyntax:
  Exclude:
    - app/views/**/*
    - app/controllers/**/*
    - app/helpers/**/*
    - app/mailers/**/*
    - app/models/**/*
    - app/poros/**/*
    - spec/**/*
```

When I run rubocop with these settings, I'm getting some really erratic behavior. Right now it is reporting every violation everywhere, without regard to the excludes. I was running it with `rubocop -a` and it would report back that 130 or so violations were fixed, when in fact it had fixed every violation in the app, again without regard to the excludes. I can't seem to get back to that state now, though. I've been trying some different patterns and getting different results.
